File: backend/merp/src/main/java/dev/mateorossello/merp/AccountingModule/DTOs/AccountOutput.java
package dev.mateorossello.merp.AccountingModule.DTOs;

import dev.mateorossello.merp.AccountingModule.Models.AccountType;

public record AccountOutput(
    Long id,

    Long parentAccountId,

    String code,

    AccountType type,

    String name,

    String description,

    Boolean receiveBalance,

    Boolean state
) {}

File: frontend/src/types/accounting/AccountOutput.ts
export const ACCOUNT_TYPES = [
  "ACTIVO",
  "PASIVO",
  "PATRIMONIO_NETO",
  "RESULTADOS_POSITIVOS",
  "RESULTADOS_NEGATIVOS",
] as const;

export type AccountType = (typeof ACCOUNT_TYPES)[number];

export interface AccountOutput {
  id: number;
  parentAccountId: number | null;
  code: string;
  type: AccountType;
  name: string;
  description: string;
  receiveBalance: boolean;
  state: boolean;
}
